### PR TITLE
[LS] Refactor language client capability usages to honor optional capabilities

### LINKED ISSUE #31948: Language server does not honour optional client fields' availability before accessing
**Description:**
Consider the following 
https://github.com/ballerina-platform/ballerina-lang/blob/v2.0.0-beta.2.1/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaLanguageServer.java#L140

Here, we check for the rename operation's availability even though the client can avoid setting it and eventually this can lead to NPEs. We have to check for the other places where we check for the optional fields and revamp them

**Affected Versions:**
SwanLake-Beta2 at least

### PR DESCRIPTION
**Pull request by malinthar**
## Purpose
$subject to avoid NPE when accessing optional client capabilities. 

Fixes #31948 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/contexts/RenameContextImpl.java [hunk 1]
```diff
@@ -45,8 +45,9 @@
         super(operation, fileUri, wsManager, cursorPos, serverContext);
         this.clientCapabilities = clientCapabilities;
         this.params = params;
-        this.honorsChangeAnnotations = clientCapabilities.getTextDocCapabilities()
-                .getRename().getHonorsChangeAnnotations();
+        this.honorsChangeAnnotations = clientCapabilities.getTextDocCapabilities().getRename() != null &&
+                clientCapabilities.getTextDocCapabilities()
```
Comment (IMS94) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/contexts/RenameContextImpl.java:49]: Shall we reformat this into a single line:
```suggestion
                clientCapabilities.getTextDocCapabilities().getRename().getHonorsChangeAnnotations();
```

**Review by IMS94 — APPROVED** [review]
LGTM! There's a minor comment.

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/contexts/RenameContextImpl.java (hunk 1 above)
Comment (malinthar) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/contexts/RenameContextImpl.java]: Done

**Review by nadeeshaan — APPROVED** [review]
LGTM

**Pull request merged**